rust-error-handling-book/structure.md

```diff
@@ -0,0 +1,82 @@
+# Book
+
+## Introduction
+
+- General introduction (for inspiration, can look at these resources)
+  - [Rust CLI Book](https://rust-cli.github.io/book/index.html)
+  - [Cargo Book](https://doc.rust-lang.org/cargo/index.html)
+  - [The book](https://doc.rust-lang.org/book/ch00-00-introduction.html)
+- Purpose of the book
+  - To give an introduction to error handling in Rust
+  - To give practical solutions to error handling design
+    - Discussing different design decisions and tradeoffs
+  - Share best practices for error handling in different scenarios
+
+## Introduction to error handling
+
+- What is error handling
+  - Jane - [Error Handling is not all about errors](https://youtu.be/WzC7H83aDZc) (from minute 3:30).
+    - Defining errors
+    - Propagating errors and gathering context
+    - Reacting errors
+    - Discarding errors
+    - Reporting errors and gathered context
+- What is the purpose of errors?
+  - Communication
+    - All about expressing intent to the developer i.e this was supposed to happen but it failed and
```

Comment:
Also for the user, and being able to moderate the information that you show to devs vs the info you show to users is a huge part of error handling design.